ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,81 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+ULONG
+IopValidateID(IN PWCHAR Id,
+              IN SIZE_T MaxIdLen,
+              IN ULONG MaxSeparators,
+              IN BOOLEAN IsMultiSz)
+{
+   PWCHAR PtrSymbol;
+   ULONG_PTR StringEnd;
+   WCHAR Symbol;
+   ULONG SeparatorsCount = 0;
+   PWCHAR PtrPrevSymbol = NULL;
+
+   PAGED_CODE();
+
+   StringEnd = (ULONG_PTR)Id + MaxIdLen * sizeof(WCHAR);
+
+   for (PtrSymbol = Id; (ULONG_PTR)PtrSymbol < StringEnd; PtrSymbol++)
```

Comment:
You can directly compare `PtrSymbol < StringEnd` without the cast.

Reply:
ok